dali/operators/imgcodec/decoder.cc

```diff
@@ -0,0 +1,411 @@
+
+The output of the decoder is in the *HWC* layout.
+
+Supported formats: JPG, BMP, PNG, TIFF, PNM, PPM, PGM, PBM, JPEG 2000, WebP.
+
+.. note::
+  JPEG 2000 region-of-interest (ROI) decoding is not accelerated on the GPU, and will use
+  a CPU implementation regardless of the selected backend. For a GPU accelerated implementation,
+  consider using separate ``decoders.image`` and ``slice`` operators.)code")
+  .NumInput(1, 3)
+  .NumOutput(1)
+  .AddParent("ImgcodecDecoderAttr")
+  .AddParent("SliceAttr")
+  .InputDox(0, "data", "TensorList", R"code(Batch that contains the input data.)code")
+  .InputDox(1, "anchor", "1D TensorList of float or int",
+            R"code(Input that contains normalized or absolute coordinates for the starting
+point of the slice (x0, x1, x2, …).
+
+Integer coordinates are interpreted as absolute coordinates, while float coordinates can be
+interpreted as absolute or relative coordinates, depending on the value of
+``normalized_anchor``.)code")
+  .InputDox(2, "shape", "1D TensorList of float or int",
+            R"code(Input that contains normalized or absolute coordinates for the dimensions
+of the slice (s0, s1, s2, …).
+
+Integer coordinates are interpreted as absolute coordinates, while float coordinates can be
+interpreted as absolute or relative coordinates, depending on the value of
+``normalized_shape``.)code");
+
+
+DALI_SCHEMA(experimental__decoders__ImageRandomCrop)
+  .DocStr(R"code(Decodes images and randomly crops them.
+
+The cropping window's area (relative to the entire image) and aspect ratio can be restricted to
+a range of values specified by ``area`` and ``aspect_ratio`` arguments, respectively.
+
+When possible, the operator uses the ROI decoding APIs (for example, *libjpeg-turbo* and *nvJPEG*)
+to reduce the decoding time and memory usage. When the ROI decoding is not supported for a given
+image format, it will decode the entire image and crop the selected ROI.
+
+The output of the decoder is in *HWC* layout.
+
+Supported formats: JPG, BMP, PNG, TIFF, PNM, PPM, PGM, PBM, JPEG 2000, WebP.
+
+.. note::
+  JPEG 2000 region-of-interest (ROI) decoding is not accelerated on the GPU, and will use
+  a CPU implementation regardless of the selected backend. For a GPU accelerated implementation,
+  consider using separate ``decoders.image`` and ``random_crop`` operators.
+
+.. note::
+  EXIF orientation metadata is used to rectify the image.)code")
+  .NumInput(1)
+  .NumOutput(1)
+  .AddParent("ImgcodecDecoderAttr")
+  .AddParent("RandomCropAttr");
+
+
+bool ImgcodecHostDecoder::SetupImpl(std::vector<OutputDesc> &output_descs,
+                            const workspace_t<CPUBackend> &ws) {
+  SetupShapes(spec_, ws, output_descs, ws.GetThreadPool());
+  return true;
+}
+
+void ImgcodecHostDecoder::RunImpl(workspace_t<CPUBackend> &ws) {
+  const auto &input = ws.Input<CPUBackend>(0);
+  auto &output = ws.Output<CPUBackend>(0);
+  output.SetLayout("HWC");
+
+  auto decoder = GetDecoderInstance();
```

Comment:
I prefer to keep *, & - just to make it a tiny bit easier to read.
```suggestion
  auto *decoder = GetDecoderInstance();
```